### PR TITLE
configuration_manual/service_configuration - Add documentation for service listener type configuration.

### PR DESCRIPTION
**Pull request by stephanbosch**
JIRA: DOV-5667

### CODE REVIEW

File: source/configuration_manual/service_configuration.rst [hunk 1]
```diff
@@ -118,6 +118,10 @@ path
 ^^^^
 Path to the file, relative to ``base_dir`` setting. This is also used as the section name.
 
+type (v3.0+)
+^^^^^^^^^^^^
```
Comment (sirainen) [inline source/configuration_manual/service_configuration.rst:122]:
```
type
^^^

.. versionadded:: v2.4.0;v3.0.0
```

File: source/configuration_manual/service_configuration.rst [hunk 2]
```diff
@@ -187,12 +195,12 @@ auth
...
    * **login**: Starts a two phase user login by performing authenticating (same as`client` type). Used by login processes.
    * **master**: Finishes the two phase user login by performing a userdb lookup (similar to "userdb" type). Used by post-login processes (e.g. imap, pop3).
 
-With UNIX listeners the client type is selected based on the filename after the last ``-`` in the filename. For example ``anything-userdb`` is of `userdb` type. The default type is `client` for inet listeners and unrecognized UNIX listeners. You can add as many client and userdb listeners as you want (and you probably shouldn't touch the login/master listeners).
+For Dovecot versions past v3.0, the listener type is configured explicitly using the **type** field. For older versions the client type is selected based on the (file)name after the last ``-`` in the name. For example ``anything-userdb`` is of `userdb` type. The default type is `auth` for inet listeners and unrecognized UNIX listeners. You can add as many client and userdb listeners as you want (and you probably shouldn't touch the login/master listeners).
```
Comment (sirainen) [inline source/configuration_manual/service_configuration.rst:203]: `.. versionchanged:: v2.4.0;v3.0.0 The listener type is ..`

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/configuration_manual/service_configuration.rst (hunk 1 above)
Comment (stephanbosch) [reply source/configuration_manual/service_configuration.rst]: changed

File: source/configuration_manual/service_configuration.rst (hunk 2 above)
Comment (stephanbosch) [reply source/configuration_manual/service_configuration.rst]: changed

**Review by sirainen — APPROVED** [review]

**Review by sirainen — APPROVED** [review]

**Pull request merged**